src/components/shared/grid-block.json

```diff
@@ -0,0 +1,96 @@
+{
+  "collectionName": "components_shared_grid_blocks",
+  "info": {
+    "displayName": "gridBlock",
+    "description": ""
+  },
+  "options": {},
+  "attributes": {
+    "heading": {
+      "type": "string"
+    },
+    "description": {
+      "type": "text"
+    },
+    "color": {
+      "type": "enumeration",
+      "enum": [
+        "white",
+        "black",
+        "background",
+        "primary",
+        "secondary",
+        "tertiary",
+        "quaternary",
+        "quinary"
+      ]
+    },
+    "bgColor": {
+      "type": "enumeration",
+      "enum": [
+        "white",
+        "black",
+        "background",
+        "primary",
+        "secondary",
+        "tertiary",
+        "quaternary",
+        "quinary"
+      ]
+    },
+    "linkColor": {
+      "type": "enumeration",
+      "enum": [
+        "white",
+        "black",
+        "background",
+        "primary",
+        "secondary",
+        "tertiary",
+        "quaternary",
+        "quinary"
+      ]
+    },
+    "linkBgColor": {
+      "type": "enumeration",
+      "enum": [
+        "white",
+        "black",
+        "background",
+        "primary",
+        "secondary",
+        "tertiary",
+        "quaternary",
+        "quinary"
+      ]
+    },
+    "linkText": {
+      "type": "string"
+    },
+    "linkUrl": {
+      "type": "string"
+    },
+    "linkIcon": {
+      "type": "media",
+      "multiple": false,
+      "required": false,
+      "allowedTypes": [
+        "images"
+      ]
+    },
+    "linkTitle": {
+      "type": "string"
+    },
+    "linkNewTab": {
```

Comment:
Here is he problem with link. You use "link" prefix before several entities, it's a signal to have link as a separate entity 
```
link: {
    icon: {
        type: media;
        ...
   },
   url: {
       type: string;
   }
}
```

By the way, Link shouldn't have a separate background. Also, I don't think that icon should be togglable (let's clarify it with Iryna).

Reply:
You absolutely right saying that the link should be a separate entity, but Strapi restricts nesting to only two levels, therefore, I had use this approach because of this limitation. I don't like it myself, but we can't seem to change it, even if I make direct nesting in the code itself, Strapi would throw error.

Reply:
As it comes for the link background, it is an optional field and we can ignore and not request it, or remove it from the data structure. The same with the icon, we can remove it.

Reply:
Well, it's a big question mark why do we use Strapi? Who has made this decision?
Ok, just a moment of anger. Let's leave it as is then if we have no choice.